trie/nodeset.go

```diff
@@ -18,52 +18,167 @@ package trie
 
 import (
 	"fmt"
+	"reflect"
+	"strings"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
 )
 
 // memoryNode is all the information we know about a single cached trie node
 // in the memory.
 type memoryNode struct {
-	hash common.Hash // Node hash, computed by hashing rlp value
-	size uint16      // Byte size of the useful cached data
-	node node        // Cached collapsed trie node, or raw rlp data
+	hash common.Hash // Node hash, computed by hashing rlp value, empty for deleted nodes
+	size uint16      // Byte size of the useful cached data, 0 for deleted nodes
+	node node        // Cached collapsed trie node, or raw rlp data, nil for deleted nodes
+}
+
+// memoryNodeSize is the raw size of a memoryNode data structure without any
+// node data included. It's an approximate size, but should be a lot better
+// than not counting them.
+// nolint:unused
+var memoryNodeSize = int(reflect.TypeOf(memoryNode{}).Size())
+
+// memorySize returns the total memory size used by this node.
+// nolint:unused
+func (n *memoryNode) memorySize(key int) int {
+	return int(n.size) + memoryNodeSize + key
+}
+
+// rlp returns the raw rlp encoded blob of the cached trie node, either directly
+// from the cache, or by regenerating it from the collapsed node.
+// nolint:unused
+func (n *memoryNode) rlp() []byte {
+	if node, ok := n.node.(rawNode); ok {
+		return node
+	}
+	enc, _ := rlp.EncodeToBytes(n.node)
```

Comment:
hm Could u track error here ?

Reply:
I remember they say that EncodeToBytes is expected to never return error, since we can always encode anything, and many later updates also do not track error here. Not sure if it is needed.

Reply:
hm when I search rlp.EncodeToBytes, I see they checks for everytimes, Fatal or even panic incase getting error, for our case I think we just need to track Error here. At least we had some thing for tracing  🤔

Reply:
Hmm okay I agree, error checking added